### PR TITLE
Add multi-write stats collector 

### PR DESCRIPTION
**Pull request by JohnlNguyen**
## PR Summary

Adding a multi-write stats collector to write to multiple stats-D hosts. It wraps a set of datadog collectors.

* **Type:** Features
* **Intended Change Level:** minor

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

Does the "Intended Change Level" above match the actual behavior of this PR?
Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

**Comment by JohnlNguyen** [comment]
@wcharczuk @chotiwat can you guys take a look

File: stats/multicollector.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package stats
+
+import (
+	"time"
+)
+
+// MultiCollector is a multi-write stats collector
+type MultiCollector interface {
```
Comment (wcharczuk) [inline stats/multicollector.go]: why can't we re-use the existing interface and just swap out the type that implements it?

File: stats/stats_collector.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/blend/go-sdk/ex"
+)
+
+// type assertion that StatsCollector implements MultiCollector
+var (
+	_ MultiCollector = (*StatsCollector)(nil)
+)
+
+// StatsCollector is a class that wraps a set of statsd collectors
+type StatsCollector struct {
```
Comment (wcharczuk) [inline stats/stats_collector.go]: I would rename this `MultiCollector` and unless it needs to have extra state on it, just type alias `type MultiCollector []Collector`

File: stats/multicollector.go (hunk 1 above)
Comment (JohnlNguyen) [reply stats/multicollector.go]: done

File: stats/stats_collector.go (hunk 2 above)
Comment (JohnlNguyen) [reply stats/stats_collector.go]: done

File: stats/multicollector.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	"time"
+
+	"github.com/blend/go-sdk/ex"
+)
+
+// MultiCollector is a class that wraps a set of statsd collectors
+type MultiCollector []Collector
+
+// NewMultiCollector returns a new multi-collector from a set of collectors
+func NewMultiCollector(collectors ...Collector) (MultiCollector, error) {
```
Comment (wcharczuk) [inline stats/multicollector.go]: i honestly wouldn't worry about this constructor; the `range` operator will handle nil or empty inputs fine

**Comment by wcharczuk** [comment]
looks like there is still untested code that codecov is complaining about; probably need to add more tests. you can look at the codecov report or the `coverage.html` that is generated by running `make cover` locally

File: stats/mock_collector.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ var (
 func NewMockCollector() *MockCollector {
 	return &MockCollector{
 		Events: make(chan MockMetric),
+		Errors: make(chan error, 30),
```
Comment (wcharczuk) [inline stats/mock_collector.go]: if the `Events` chan does't have a length the errors chan probably shouldn't either.

Comment (JohnlNguyen) [reply stats/mock_collector.go]: done

**Pull request merged**